consensus/state.go

```diff
@@ -362,6 +362,14 @@ func (cs *ConsensusState) OnStart() error {
 		// let's go for it anyways, maybe we're fine
 	}
 
+	// If the latest block was applied in the tmsp handshake,
```

Comment:
yep, we want the cswal to actually handle all the cases: https://github.com/tendermint/tendermint/issues/352